### PR TITLE
Verifying a number of NRT bugs are fixed

### PR DESCRIPTION
**Pull request by jaredpar**
Found a number of NRT bugs which are now fixed and added the appropriate verification. In the EE tests it was mostly ensuring the tests were run with the latest language version (now the default).

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-compiler for review

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 1]
```diff
@@ -74919,6 +74919,29 @@ A M()
...
     }
 } 
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [WorkItem(28324, "https://github.com/dotnet/roslyn/issues/28324")]
+        [Fact]
+        public void CopyTypeCustomModifiersWithGenericOverridenMethod()
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >Overriden [](start = 54, length = 9)

Overridden?

Comment (jaredpar) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: At some point I should probably appologize for all the typos I've made you see over the years ...

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 2]
```diff
@@ -74919,6 +74919,29 @@ A M()
         return s; 
     }
 } 
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [WorkItem(28324, "https://github.com/dotnet/roslyn/issues/28324")]
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >https://github.com/dotnet/roslyn/issues/28324 [](start = 26, length = 45)

Looks like we have a test already: `NonNullTypes_GenericOverriddenMethod_ValueType`

Comment (jaredpar) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: Okay. I will delete this and add the `WorkItem` notation to that one.

File: src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/ExpressionCompilerTests.cs [hunk 3]
```diff
@@ -3787,42 +3787,35 @@ static void M()
 }";
             var assemblyName = ExpressionCompilerUtilities.GenerateUniqueName();
             // https://github.com/dotnet/roslyn/issues/30031: error CS0101: The namespace 'System.Runtime.CompilerServices' already contains a definition for 'NullableAttribute'
```
Comment (cston) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/ExpressionCompilerTests.cs]: >// https://github.com/dotnet/roslyn/issues/30031 [](start = 12, length = 48)

Comment can be removed.

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs (hunk 2 above)
Comment (jaredpar) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: Hhu. It's already tagged with that work item number. I'll just remove this test then.

File: src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/UsingDebugInfoTests.cs [hunk 4]
```diff
@@ -228,7 +228,7 @@ void M()
 }
 ";
             // https://github.com/dotnet/roslyn/issues/30030: C#8 projects require System.Attribute.
```
Comment (jaredpar) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/UsingDebugInfoTests.cs]: Will delete this comment.

**Pull request merged**